Pulled EXCEPTION_MESSAGE up to module constant

sam-smo opened the pull request:
Fixes issue #28 

I tried making test_two_tables fail by changing the expected sql, but no exception was thrown. I was able to verify the change worked by making test_select_expression fail. Do you know why test_two_tables didn't fail?

klahnakoski commented:
which `test_two_tables`?  there are three of them.  What did you change he SQL to?

sam-smo commented:
Thank you for explaining. I will be able to look into it more and get back
to you when I am at my computer later.

On Thu, Jan 3, 2019 at 7:22 AM Kyle Lahnakoski <notifications@github.com>
wrote:

> The verify_formatting function is transforming expected_json -> SQL ->
> JSON, and never uses the expected_sql. This was done to make the tests
> "more stable"; they would still work in face of whitespace formatting
> changes. We assume the SQL is correct if it parses to the original JSON.
>
> We can certainly improve the verify_formatting function so it uses the
> expected_sql. It will probably demand we fix a number of formatting tests
> because the whitespace is wrong, it may also catch an error in the
> parser/formatter. But that is for another issue/pull request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-451128684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApgUdo0p_4CVD5zuwxs6yJfaegmeE6nsks5u_fYcgaJpZM4Zneza>
> .
>
-- 
Sam Smouha
samsmouha.com

klahnakoski commented:
Thank you!   Looks good now!

sam-smo commented:
I think I only see one of the test_two_tables (line 57).

I changed the code in a few places to run thetests. In line 59,  'XYZZY'
changed to 'SAM'.
In line 64, 'SYS.XYZZY' to 'SAM'.
And I think the last I tried was in line 70, 'dual' to 'SAM'.


Sam Smouha
samsmouha.com


On Wed, Jan 2, 2019 at 11:33 PM Kyle Lahnakoski <notifications@github.com>
wrote:

> which test_two_tables? there are three of them. What did you change he
> SQL to?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-451054148>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApgUdgIDvSIw239-vkPf5VDpdukAKsSWks5u_YgugaJpZM4Zneza>
> .
>

sam-smo commented:
Are there any other bugs or issues you would like to work on together?

On Sun, Jan 6, 2019 at 8:52 PM Sam Smouha <samsmouha@gmail.com> wrote:

> I think I only see one of the test_two_tables (line 57).
>
> I changed the code in a few places to run thetests. In line 59,  'XYZZY'
> changed to 'SAM'.
> In line 64, 'SYS.XYZZY' to 'SAM'.
> And I think the last I tried was in line 70, 'dual' to 'SAM'.
>
>
> Sam Smouha
> samsmouha.com
>
>
> On Wed, Jan 2, 2019 at 11:33 PM Kyle Lahnakoski <notifications@github.com>
> wrote:
>
>> which test_two_tables? there are three of them. What did you change he
>> SQL to?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-451054148>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ApgUdgIDvSIw239-vkPf5VDpdukAKsSWks5u_YgugaJpZM4Zneza>
>> .
>>
> --
Sam Smouha
samsmouha.com

klahnakoski commented:
Yes, please!   In this repo there are a number of `@skip`ped tests.  Please have a look and see which ones you would like to fix.

then there is the larger concern with the pyparsing module (https://github.com/mozilla/moz-sql-parser/issues/41 ):  I did not estimate how much work it is; breaking down the problem should be the first step.

There are a number of other projects, which I have not given thought to yet; I will email you as I prepare them. It will not be this week.

sam-smo commented:
Ok thanks. I will take a look tonight to see if I can start figuring
anything out.

On Tue, Jan 8, 2019 at 3:30 PM Kyle Lahnakoski <notifications@github.com>
wrote:

> Yes, please! In this repo there are a number of @skipped tests. Please
> have a look and see which ones you would like to fix.
>
> then there is the larger concern with the pyparsing module (#41
> <https://github.com/mozilla/moz-sql-parser/issues/41> ): I did not
> estimate how much work it is; breaking down the problem should be the first
> step.
>
> There are a number of other projects, which I have not given thought to
> yet; I will email you as I prepare them. It will not be this week.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-452440946>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApgUdp9uTz4Epy2bH4E1vlEaFaWxg-lkks5vBP_ugaJpZM4Zneza>
> .
>
-- 
Sam Smouha
samsmouha.com

klahnakoski commented:
feel free to open a PR with "WIP" (work in progress) prefix: your code need not work, and we can discuss ideas or questions there.

sam-smo commented:
Ok, sounds great. I will be able to do that tonight. Thank you.


Sam Smouha
samsmouha.com


On Wed, Jan 9, 2019 at 5:42 PM Kyle Lahnakoski <notifications@github.com>
wrote:

> feel free to open a PR with "WIP" (work in progress) prefix: your code
> need not work, and we can discuss ideas or questions there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-452857110>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApgUdoKa1wvuWA9Hj6mphedoU6p8Ct7lks5vBmBwgaJpZM4Zneza>
> .
>

sam-smo commented:
Where do I add that WIP prefix? I just did another pull request, but not
sure I entered the correct things under "base" and "compare."
Thanks

Sam Smouha
samsmouha.com


On Wed, Jan 9, 2019 at 5:42 PM Kyle Lahnakoski <notifications@github.com>
wrote:

> feel free to open a PR with "WIP" (work in progress) prefix: your code
> need not work, and we can discuss ideas or questions there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-452857110>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApgUdoKa1wvuWA9Hj6mphedoU6p8Ct7lks5vBmBwgaJpZM4Zneza>
> .
>

klahnakoski commented:
Each pull request has a title, put the "WIP" in the title.

klahnakoski commented:
Do not worry about making mistakes.  "base" will be the `dev` branch of this repo (should be the default).  The compare will be the branch you are working with on your repo.

sam-smo commented:
Sorry for the delay. I am working on this today.


Sam Smouha
samsmouha.com


On Thu, Jan 10, 2019 at 11:14 PM Kyle Lahnakoski <notifications@github.com>
wrote:

> Do not worry about making mistakes. "base" will be the dev branch of this
> repo (should be the default). The compare will be the branch you are
> working with on your repo.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/moz-sql-parser/pull/44#issuecomment-453372668>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApgUdnnLreRSGmtjxbjLydVnQ9VyuOt7ks5vCA-2gaJpZM4Zneza>
> .
>
